src/parser/mod.rs

```diff
@@ -17225,66 +17229,84 @@ impl<'a> Parser<'a> {
 
             self.expect_keyword_is(Keyword::THEN)?;
 
+            macro_rules! not_allowed {
```

Comment:
maybe we can keep this inlined as before, alternative would be to pull it out to a function but that might not be worth it. But in generally we'd ideally avoid custom macros where possible since those are harder to maintain